### ISSUE TITLE
Eliminate crashes with ebinds

### ISSUE THREAD

**Issue by ranjitjhala** [ucsd-progsys/liquid-fixpoint#364]
The below crashes with `subst: EXISTS (without disjoint binds)"`

```
fixpoint "--eliminate=all"

// This file is SAFE with --eliminate=none
// but both qualifiers are needed
// qualif Auto(v : int, m : int): (v = (m + 2))
// qualif Auto(v : int, x : int): (v = x)

// However, --eliminate=some marks this UNSAFE
// but this can be fixed by removing cons 3

// Constraint 3 is only needed to prevent the *sharing* optimization

bind  1 m    : {v : int | true }
ebind 2 n    : { int }

constraint:
  env [1; 2]
  lhs {v : int | $k[vk := v] }
  rhs {v : int | v = n       }
  id 4 tag []

constraint:
  env [1]
  lhs {v : int | v = m + 1   }
  rhs {v : int | $k[vk := v] }
  id 5 tag []

constraint:
  env [2]
  lhs {v : int | true     }
  rhs {v : int | $k[vk := v] }
  id 3 tag []

wf:
  env [1]
  reft {vk : int | [$k]}
```

**Comment by atondwal** [comment]
Here's an acyclic version:

`[ebind-kvar-chain.fq]`
```
fixpoint "--eliminate=some"

// solve WITHOUT these qualifiers
// qualif Auto(v : int, m : int): (v = (m + 2))
// qualif Auto(v : int, x : int): (v = x)

ebind 15 n    : { int }

bind  16 m    : { _ : int | true  }
bind  18 one  : { v : int | v = 1 }

constraint:
  env [15]
  lhs {v3 : int | v3 = n       }
  rhs {v3 : int | $k2[vk2:=v3] }
  id 3 tag []

constraint:
  env [15]
  lhs {v4 : int | $k1[vk1:=v4] }
  rhs {v4 : int | v4 = n       }
  id 4 tag []

constraint:
  env [16; 18]
  lhs {v5 : int | v5 = m + 1   }
  rhs {v5 : int | $k1[vk1:=v5] }
  id 5 tag []

constraint:
  env [16; 18]
  lhs {v6 : int | $k2[vk2:=v6] }
  rhs {v6 : int | v6 = m + 1   }
  id 6 tag []

wf:
  env [16]
  reft {vk1 : int   | [$k1]}

wf:
  env [16]
  reft {vk2 : int   | [$k2]}
```

**Comment by atondwal** [comment]
This is fixed by PR #369
